src/pedal/config.rs

```diff
@@ -0,0 +1,50 @@
+use crate::{pedal, Pedal as PedalTrait, Result};
+use cpal::StreamConfig;
+use serde::Deserialize;
+use Pedal::*;
+
+#[derive(Debug, Deserialize)]
+pub struct Config {
+    pedals: Vec<Pedal>,
+}
+
+impl Config {
+    pub fn from(yaml: &str) -> Result<Config> {
+        Ok(serde_yaml::from_str(yaml)?)
+    }
+
+    pub fn default() -> Self {
+        Self {
+            pedals: vec![Transparent],
+        }
+    }
+
+    pub fn to_pedals(&self, stream_config: &StreamConfig) -> Result<Vec<pedal::Boxed>> {
+        self.pedals
+            .iter()
```

Comment:
Rather than directly exposing `map`, etc. on `Vec`s and other collection types, they all provide `iter` (and some related functions) for creating a lazy iterator.